core/Types.h

```diff
@@ -75,13 +75,22 @@ class Types final {
     /** Lower upper bound: the narrowest type that is supper type of both t1 and t2 */
     static TypePtr any(Context ctx, const TypePtr &t1, const TypePtr &t2);
 
-    /** is every instance of  t1 an  instance of t2? */
-    static bool isSubTypeUnderConstraint(Context ctx, TypeConstraint &constr, const TypePtr &t1, const TypePtr &t2);
+    /**
+     * is every instance of  t1 an  instance of t2?
+     *
+     * The parameter `allowUntyped` controls whether or not `T.untyped` is
+     * considered to be a super type or subtype of all other types */
+    static bool isSubTypeUnderConstraint(Context ctx, TypeConstraint &constr, const bool allowUntyped,
```

Comment:
Thoughts on using an `enum class` instead of a boolean?

Reply:
e.g., like this

https://github.com/sorbet/sorbet/pull/1472/files#diff-3416fec1d98d0d7b3662ee1e5f4ec64cR156

Reply:
I like it. I'll make that change :)

Reply:
I might even go so far as to make it like

```cpp
enum class UntypedMode {
  AlwaysCompatible = 1,
  AlwaysIncompatible = 2,
};
```

Reply:
I made this change, thanks @jez!